refbox/src/app/mod.rs

```diff
@@ -961,10 +961,16 @@ impl Application for RefBoxApp {
                 } else {
                     let mut tm = self.tm.lock().unwrap();
                     let now = Instant::now();
-                    tm.add_score(color, 0, now);
-                    let snapshot = tm.generate_snapshot(now).unwrap(); // TODO: Remove this unwrap
-                    std::mem::drop(tm);
-                    self.apply_snapshot(snapshot);
+                    if tm.current_period() == GamePeriod::SuddenDeath {
+                        tm.stop_clock(now).unwrap();
+                        let mut scores = tm.get_scores();
+                        scores[color] = scores[color].saturating_add(1);
+
+                        self.app_state = AppState::ConfirmScores(scores);
+                    } else {
+                        tm.add_score(color, 0, now);
+                        self.app_state = AppState::MainPage;
+                    }
                 }
             }
```

Comment:
The `trace!("AppState changed to {:?}", self.app_state);` line above should be moved between these two lines so that a change in `self.app_state` is always logged